connectivity/cellular/source/framework/RIL/RIL_CellularContext.cpp

```diff
@@ -0,0 +1,1164 @@
+    }
+
+    delete _data_call_addresses;
+    _data_call_addresses = nullptr;
+}
+
+nsapi_error_t RIL_CellularContext::set_blocking(bool blocking)
+{
+    nsapi_error_t err = NSAPI_ERROR_OK;
+    tr_info("RIL_CellularContext set blocking %d", blocking);
+    _is_blocking = blocking;
+    return err;
+}
+
+NetworkStack *RIL_CellularContext::get_stack()
+{
+    if (!_stack) {
+        _stack = (NetworkStack *)&LWIP::get_instance();
+    }
+    return _stack;
+}
+
+nsapi_error_t RIL_CellularContext::get_ip_address(SocketAddress *address)
+{
+    if (_stack_interface) {
+        return _stack_interface->get_ip_address(address);
+    } else {
+        return NSAPI_ERROR_NO_CONNECTION;
+    }
+}
+
+char *RIL_CellularContext::get_interface_name(char *interface_name)
+{
+    if (_stack_interface) {
+        return _stack_interface->get_interface_name(interface_name);
+    } else if (_pdp_type == NON_IP_PDP_TYPE) {
+        sprintf(interface_name, "ni%d", _cid);
+        return interface_name;
+    } else {
+        return nullptr;
+    }
+}
+
+void RIL_CellularContext::attach(mbed::Callback<void(nsapi_event_t, intptr_t)> status_cb)
+{
+    _status_cb = status_cb;
+}
+
+nsapi_error_t RIL_CellularContext::connect()
+{
+    tr_info("CellularContext connect");
+    ScopedLock<rtos::Mutex> lock(_api_mutex);
+
+    if (_is_connected) {
+        return NSAPI_ERROR_IS_CONNECTED;
+    }
+
+    if (_is_running) {
+        return NSAPI_ERROR_BUSY;
+    }
+
+    _is_running = true;
+    _final_op = OP_CONNECT;
+    nsapi_error_t err = _device->set_sim_ready();
+    _cb_data.error = check_operation(err, OP_SIM_READY);
+
+    if (_is_blocking) {
+        if (_cb_data.error == NSAPI_ERROR_OK || _cb_data.error == NSAPI_ERROR_ALREADY) {
+            do_connect_with_retry();
+        }
+        _is_running = false;
+    } else {
+        if (_cb_data.error == NSAPI_ERROR_ALREADY) {
+            // device is already attach state.
+            int id = _device->get_queue()->call_in(0, this, &RIL_CellularContext::do_connect_with_retry);
+            if (id == 0) {
+                _is_running = false;
+                _final_op = OP_INVALID;
+                _current_op = OP_INVALID;
+                return NSAPI_ERROR_NO_MEMORY;
+            }
+            return NSAPI_ERROR_OK;
+        }
+    }
+
+    return _cb_data.error;
+}
+
+void RIL_CellularContext::do_connect_with_retry()
+{
+    CellularContext::do_connect_with_retry();
+}
+
+nsapi_error_t RIL_CellularContext::do_initial_attach()
+{
+    ScopedLock<rtos::Mutex> lock(_api_mutex);
+
+    _set_registration_complete = false;
+    nsapi_error_t ret = NSAPI_ERROR_OK;
+
+    rtos::Mutex *cond_mutex = new rtos::Mutex;
+    rtos::ConditionVariable *cond_var = new rtos::ConditionVariable(*cond_mutex);
+
+    cond_mutex->lock();
+
+    RILAdaptation &ril = static_cast<RIL_CellularDevice *>(_device)->get_ril();
+
+    ril_token_t *token1;
+    
+    token1 = ril.send_request(RIL_REQUEST_SET_NETWORK_SELECTION_AUTOMATIC,
```

Comment:
Why manual PLMN selection has been removed?

Reply:
This PR is based on mbed os 6 to add RIL interface, the pr-S5JS100-06_RIL is based on mbed os 5, so there is no `const char *get_plmn() const` function in the CellularDevice.h
Add our principle in the adaptation process is not to change your commit code.

Reply:
Manual PLMN was a requested feature by Samsung in the first place.
Current didn't have `get_plmn` API call but it was added during cellular RIL development.

Reply:
OK, I got it and will to add it, thanks